### PR TITLE
KAA-1481: 1-thread callback executor (Java client) causes issue with "nested" callback calls

### PR DESCRIPTION
**Pull request by Serega290696**

### CODE REVIEW

**Comment by kaa-leeroy** [comment]
Can one of the admins verify this patch?

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/context/FlexibleExecutorContext.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        public FlexibleExecutorContextBuilder setCallbackThreadCountAndIdleMilliseconds(
+                int callbackThreadCount, int maxCallbackThreadsIdleMilliseconds) {
+            this.callbackThreadCount = callbackThreadCount;
+            this.maxCallbackThreadsIdleMilliseconds = maxCallbackThreadsIdleMilliseconds;
+            return this;
+        }
+
+
+        public FlexibleExecutorContext build() {
+            FlexibleExecutorContext f = new FlexibleExecutorContext(this);
```
Comment (Rib47) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/context/FlexibleExecutorContext.java]: You must check builder fields for **null** values. If any **null** value will be found you must throw an exception or  set fields to default values. The second alternative is better one.

File: client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
+package org.kaaproject.kaa.client.context;
```
Comment (Rib47) [inline client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java:17]: missed license in this file
see documentation:
https://docs.google.com/document/d/1gesi7pElMnoT5ocefV0qLooEMnZuTvTc81xSdJQGGTI

File: client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
```
Comment (Rib47) [inline client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/context/FlexibleExecutorContextTest.java:18]: missed license in this file
see documentation:
https://docs.google.com/document/d/1gesi7pElMnoT5ocefV0qLooEMnZuTvTc81xSdJQGGTI

**Review by Rib47 — CHANGES_REQUESTED** [review]
Missed licenses

**Pull request closed without merging**